File: Domains/Base/Workspace/Autotest/Angular/src/Compiler/Html/ExpansionCase.ts
import * as compiler from '@angular/compiler';

import { Node, nodeFactory } from './Node';

export class ExpansionCase implements Node {

    expression: Node[];

    value: string;

    constructor(public node: compiler.ExpansionCase) {
        this.expression = node.expression.map((v) => nodeFactory(v));
        this.value = node.value;
    }

    public toJSON(): any {

        const { expression, value } = this;

        return {
            kind: 'expansionCase',
            expression,
            value
        };
    }
}

File: Domains/Base/Workspace/Autotest/Model/Export/Base/Html/ExpansionCase.cs
// <copyright file="ExpansionCase.cs" company="Allors bvba">
// Copyright (c) Allors bvba. All Rights Reserved.
// Licensed under the LGPL v3 license.
// </copyright>

using Autotest.Angular;

namespace Autotest.Html
{
    using System.Linq;

    using Newtonsoft.Json.Linq;

    public partial class ExpansionCase : INode
    {
        public ExpansionCase(JToken json, Template template, INode parent)
        {
            this.Json = json;
            this.Template = template;
            this.Parent = parent;
        }

        public INode[] Expression { get; set; }

        public JToken Json { get; }

        public Template Template { get; }

        public INode Parent { get; set; }

        public string Value { get; set; }

        public void BaseLoad()
        {
            this.Value = this.Json["value"]?.Value<string>();

            var expressionChildren = this.Json["expression"];
            this.Expression = expressionChildren != null ? expressionChildren.Select(v =>
                {
                    var node = NodeFactory.Create(v, this.Template, this);
                    node.BaseLoad();
                    return node;
                }).ToArray() : new INode[0];
        }
    }
}